### ISSUE TITLE
Add ST_QA band for Landsat C2L2 products

### ISSUE THREAD

**Issue by BastienKovac** [sertit/eoreader#99]
When using the thermal bands from Landsat C2L2 products, the band ST_QA is a critical piece of information. This band contains the uncertainty of the Surface Temperature band (ST_B10), in K. It is used to ignore outlier values, resulting for instance from passing clouds.

There are currently, as far as I'm aware, no way to retrieve this band through EOReader so it needs to be loaded manually.

It would be interesting to be able to load this band directly. I see two ways of doing this : 

1. Add the ST_QA band to the available bands for Landsat C2L2, which is probably the most straight-forward solution.

2. Return the ST_QA band alongside TIR_1 in the resulting XArray. The big downside of this approach is that it might break retro-compatibility, but I'd argue that the ST_QA band should pretty much always be used alongside the surface temperature. 

This would also allow to pass a 'max_incertitude' kwarg to the load method, automatically replacing surface temperature pixels exceeding the given incertitude threshold by NaNs. Although this might go beyond the scope of this feature request.

**Comment by BastienKovac** [comment]
I just realised the TIR_1 band from EOReader does not in fact match the B10 (Surface Temperature) band, making this issue pull double duty as B10 doesn't seem to be handled either. 
However this band is only available for L8/9 so it would probably need a specialised subclass of LandsatProduct

**Comment by remi-braun** [comment]
I thanks for this issue !

I will first try to answer the second message.
B10 is the band read for `TIR_1`, so I don't understand what makes you think the contrary 😅 Could you precise what you mean by "matching" ? Are you converting the band to BT on your side by using the Landsat CCoefficients that can be found in the metadata?

Concerning the first message:
It could be a good thing to add the possibility to load masks in a not hidden way for every satellite. This would however take time. For now, you can just load this band by using the following workaround: 
```python
c2l2_prod._read_band(c2l2_prod._get_path("ST_QA"))

**Comment by BastienKovac** [comment]
Thanks for the quick response and the workaround, this will work wonder 

About the B10 issue : 

After having a closer look at EOReader's code, I got mistaken between the `_map_bands_tm()` and `_map_bands_etm()` methods, so the band is indeed correct.

The values seem slightly off to me though, which is why I initially thought of retrieving the QA band. 

I first opened it and tried to convert it to Celsius by first applying `B10 * 0.00341802 + 149` to go from reflectance to Kelvin, as per the [product's documentation](https://d9-wret.s3.us-west-2.amazonaws.com/assets/palladium/production/s3fs-public/media/files/LSDS-1619_Landsat8-9-Collection2-Level2-Science-Product-Guide-v5.pdf) (Page 12).

If I understand correctly, the conversion between Reflectance and Kelvins is done automatically when loading the band ? However the values don't seem to match when I do the same conversion manually.

This is the result of TIR_1 - 273 (to go to Celsius) with EOReader : 

![eoreader](https://github.com/sertit/eoreader/assets/15960657/8cf8889e-f7bc-47ae-b66b-f65d1c4b3d31)

And this is the result of (B10 * 0.00341802 + 149) - 273 with Rasterio : 

![rasterio](https://github.com/sertit/eoreader/assets/15960657/e91f9a34-5b82-419e-9925-cfdb3eb35137)

Note the different values in the legends

**Comment by remi-braun** [comment]
The only process I'm doing to the BT bands are the L1 conversion (see the `_to_tb` function, following [this tutorial](https://www.usna.edu/Users/oceano/pguth/md_help/remote_sensing_course/landsat_thermal.htm))

It should be OK for L1 data, but you are right, maybe it is wrong for L2 data (to be honest, I never really use L2 data, so this may not be tested well 😓)

If I understand correctly, for L2 level, the offset and gain I should use from the MTD are the one stored in `LEVEL2_SURFACE_REFLECTANCE_PARAMETERS` and `LEVEL2_SURFACE_TEMPERATURE_PARAMETERS`, is that right ?

**Comment by BastienKovac** [comment]
There is an issue when loading the data : The metadata can't be read.

The issue comes from the band name, which is ST_10 internally whereas the expected metadata name is TEMPERATURE_MULT_BAND_ST_**B**10

**Comment by BastienKovac** [comment]
Sorry my last comment wasn't really clear. There is a problem with the metadata being read, which isn't caught in the try/except block because it's a TypeError, and not a ValueError : 

`TypeError: float() argument must be a string or a real number, not 'NoneType'`

**Comment by remi-braun** [comment]
Arf sorry, I did this too fast and I haven't noticed I wasn't testing thermal bands in my CI 🙄 
This should be OK now, could you please try again?

**Comment by BastienKovac** [comment]
Yes, all fixed. The two resulting rasters (rasterio and eoreader) are identical ! Thank you

**Comment by remi-braun** [comment]
Ok I'll close this issue as your demand for loading quality masks is openend in another issue:
- #101
